### PR TITLE
Add Simulator tutorial notebook

### PR DESCRIPTION
**Pull request by corentinlger**
## Description

- Added simulator tutorial notebook 
- Added an explanation on tutorials in the README
- Added saving functions in simulator 
- Renamed 'nve_idx' 'ent_idx'

## Related Issue (if applicable)
#28 

## How to Test

Follow the steps in the new simulator_tutorial notebook.

Check the server-client connection still works fine.

### CODE REVIEW

File: notebooks/simulator_tutorial.ipynb [hunk 1]
```diff
@@ -0,0 +1,627 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Questions : \n",
+    "\n",
+    "- Should I have more detailed explanations on every field in the initialization of states ?\n",
+    "- Do you wanna talk about neighbor maps in jax_md ?\n",
+    "- Do we talk about the physics engine (only way to control the simulation atm) ?\n",
+    "- Do we talk about braintenberg behaviors ? \n",
+    "- Do we talk about a future refactoring ? \n",
```
Comment (clement-moulin-frier) [inline notebooks/simulator_tutorial.ipynb:13]: Not necessary for now I think

File: notebooks/simulator_tutorial.ipynb [hunk 2]
```diff
@@ -0,0 +1,627 @@
...
+    "- What kind of custom scene do we create ? "
```
Comment (clement-moulin-frier) [inline notebooks/simulator_tutorial.ipynb:14]: Maybe just start with reproducing the same one as in web_interface.md, but doing it programatically.
You could first do it using the pre-implemented braitenberg behaviors, then showing how to program your own behavior (e.g. just reimplementing some of the braitenberg ones).
Then, to see if we could show how to add newborn and remove dead agents

File: notebooks/simulator_tutorial.ipynb [hunk 3]
```diff
@@ -0,0 +1,627 @@
...
+    "\n",
+    "- ***Agent State***: This element focuses on the agents in the simulation, which are primarily based on Braitenberg vehicles. It includes information about their behavior, wheel diameter, speed multipliers, maximum speed, and other agent-specific parameters that determine how they react to their environment and objects within it.\n",
+    "\n",
+    "- ***Object State***: This element provides information about the objects present in the simulation. Currently, it mainly consists of the objects' color, but it can be expanded to include additional object-specific parameters as needed.\n",
+    "\n",
+    "- ***Entity State***: This element combines information about both agents and objects, encompassing their physical parameters used in the physics simulation. It includes position, speed, mass, friction, and other relevant parameters. By grouping these parameters together, we can more easily compute and update the physical aspects of the simulation... (i.e update neighbors ... talk abt it) However, this structure might change in the future.\n",
```
[the remark this answers is not in the document]
Comment (corentinlger) [reply notebooks/simulator_tutorial.ipynb:46]: indeed I asked this in a commented line, I'll do that

**Review by clement-moulin-frier — COMMENTED** [review]
Thanks @corentinlger , a few comments in the code

Also, it will be useful to add a short description of all tutorials we already have in the README and in which order it is recommended to follow them. Maybe first the web_interface_tutorial.md ; then the quickstart_tutorial.ipynb ; and then this one

**Review by corentinlger — COMMENTED** [review]
I double checked the code and played with the simulator in interface mode and it doesn't seem to be errors. I merge the PR.

**Pull request merged**